### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by jyz523**
Name: Yang Tzu Yu
ID: 48338004

Hi, I'm submitting my code for the Improved UNet for the ISIC 2018 segmentation dataset (Task 1).

- train.py: Trains a neural network on the ISIC2018 dataset for skin lesion segmentation using the Improved U-Net architecture, custom loss functions, and learning rate adjustments, followed by a prediction phase.
- predict.py: Uses a pre-trained Improved U-Net model to predict skin lesion segmentations on the ISIC2018 dataset, displaying the original images alongside their predicted and true masks.
- modules.py: Defines the ImprovedUNet class, an enhanced U-Net neural network model with an encoder-decoder structure and additional convolutional layers for semantic segmentation tasks, optimized for GPU usage if available.
- dataset.py: Sets up the ISIC Dataset class for handling ISIC skin lesion images and their corresponding masks, including resizing and optional transformations, and defines specific transformation functions for both images and masks, ensuring compatibility with PyTorch's DataLoader.

### CODE REVIEW

**Comment by jyz523** [comment]
Update: It can reach 80% around 40 Epochs

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.85765. Predicted Mask output looks good. 
- Try to use PyTorch to construct the Improved U-Net model, but the model body is roughly a U-Net-like model(filters start with 64 and end up with 512) instead of the improved U-Net one(filters start with 16 and end up with 256). No data augmentation.
- Good coding design.
- Sufficient docstring and comment in the code, but need some references.
- Easy difficulty problem, no higher difficulty marks.

**Comment by jyz523** [comment]
> * Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
> * Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.85765. Predicted Mask output looks good.
> * Try to use PyTorch to construct the Improved U-Net model, but the model body is roughly a U-Net-like model(filters start with 64 and end up with 512) instead of the improved U-Net one(filters start with 16 and end up with 256). No data augmentation.
> * Good coding design.
> * Sufficient docstring and comment in the code, but need some references.
> * Easy difficulty problem, no higher difficulty marks.

So anything that need to improve ir order to get more marks or NO

**Comment by wangzhaomxy** [comment]
> > * Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
> > * Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.85765. Predicted Mask output looks good.
> > * Try to use PyTorch to construct the Improved U-Net model, but the model body is roughly a U-Net-like model(filters start with 64 and end up with 512) instead of the improved U-Net one(filters start with 16 and end up with 256). No data augmentation.
> > * Good coding design.
> > * Sufficient docstring and comment in the code, but need some references.
> > * Easy difficulty problem, no higher difficulty marks.
> 
> So anything that need to improve ir order to get more marks or NO

Just some comments for marking.

**Comment by jyz523** [comment]
> > > * Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
> > > * Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.85765. Predicted Mask output looks good.
> > > * Try to use PyTorch to construct the Improved U-Net model, but the model body is roughly a U-Net-like model(filters start with 64 and end up with 512) instead of the improved U-Net one(filters start with 16 and end up with 256). No data augmentation.
> > > * Good coding design.
> > > * Sufficient docstring and comment in the code, but need some references.
> > > * Easy difficulty problem, no higher difficulty marks.
> > 
> > 
> > So anything that need to improve ir order to get more marks or NO
> 
> Just some comments for marking.

oh ok Thanks, then I'm not gonna change anything with the code

**Comment by jyz523** [comment]
- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.        -> FIXED

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, but not improved UNet model only regular UNet -4
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no usage -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, move readme into project, restore repo readme -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by jyz523** [comment]
> Feedback marks possible +2 if the requested changes are made (see above).

Just saw this now, all done and fixed.
Thank you

**Comment by jyz523** [comment]
> No feedback attempt and no feedback marks granted.

Isn’t the attempt till 29th?

**Comment by wangzhaomxy** [comment]
Feedback attempt, but not successful. Please see Shakes' instruction above and do the modification works. But I will give you the feedback marks granted +2. Please try it again so that Shakes can merge your pull request.
Cheers.
